packages/webui/src/client/lib/viewPort.ts

```diff
@@ -251,41 +268,34 @@ function getRegionPosition(topElement: HTMLElement, bottomElement: HTMLElement):
 	return { top, bottom }
 }
 
-let scrollToPositionRequest: number | undefined
-let scrollToPositionRequestReject: ((reason?: any) => void) | undefined
+export const viewPortScrollingState = {
+	isProgrammaticScrollInProgress: false,
+	lastProgrammaticScrollTime: 0,
+}
```

Comment:
Do we really need to expose this object? What if we have it private to this module, but export functions like `isProgrammaticScrollInProgress()`, the same way there's a `isMaintainingFocus()`?

Reply:
Has added a cleanup.

Reply:
have added a getViewPortScrollingState()